masonry/src/widget/sized_box.rs

```diff
@@ -416,39 +410,19 @@ impl Widget for SizedBox {
 
 // --- BackgroundBrush ---
 
-impl BackgroundBrush {
-    /// Draw this brush into a provided [`PaintCtx`].
-    pub fn paint(&mut self, ctx: &mut PaintCtx, scene: &mut Scene) {
-        let bounds = ctx.size().to_rect();
-        match self {
-            Self::Color(color) => fill_color(scene, &bounds, *color),
-            Self::Gradient(grad) => scene.fill(
-                Fill::NonZero,
-                Affine::IDENTITY,
-                &*grad,
-                Some(Affine::IDENTITY),
-                &bounds,
-            ),
-            Self::PainterFn(painter) => painter(ctx),
-        }
-    }
-}
-
-impl From<Color> for BackgroundBrush {
-    fn from(src: Color) -> BackgroundBrush {
-        BackgroundBrush::Color(src)
-    }
-}
-
-impl From<Gradient> for BackgroundBrush {
-    fn from(src: Gradient) -> BackgroundBrush {
-        BackgroundBrush::Gradient(src)
-    }
-}
-
-impl<Painter: FnMut(&mut PaintCtx) + 'static> From<Painter> for BackgroundBrush {
-    fn from(src: Painter) -> BackgroundBrush {
-        BackgroundBrush::PainterFn(Box::new(src))
+/// Draw this brush into a provided [`PaintCtx`].
+fn paint(brush: &Brush, ctx: &mut PaintCtx, scene: &mut Scene) {
+    let bounds = ctx.size().to_rect();
+    match brush {
+        Brush::Solid(color) => fill_color(scene, &bounds, *color),
+        Brush::Gradient(grad) => scene.fill(
+            Fill::NonZero,
+            Affine::IDENTITY,
+            grad,
+            Some(Affine::IDENTITY),
+            &bounds,
+        ),
+        Brush::Image(image) => scene.draw_image(image, Affine::IDENTITY),
     }
```

Comment:
Why not use?
```suggestion
    scene.fill(
        Fill::NonZero,
        Affine::IDENTITY,
        brush,
        Some(Affine::IDENTITY),
        &bounds,
    );
```
I also don't really think this needs a helper function.

Reply:
Yeah, wanted to clarify that.

Reply:
Done.